src/common/interop/CommonManaged.cpp

```diff
@@ -0,0 +1,39 @@
+#include "pch.h"
+#include "CommonManaged.h"
+#include "CommonManaged.g.cpp"
+#include <common/version/version.h>
+#include "../../modules/videoconference/VideoConferenceShared/MicrophoneDevice.h"
+#include "../../modules/videoconference/VideoConferenceShared/VideoCaptureDeviceList.h"
+
+namespace winrt::PowerToys::Interop::implementation
+{
+    hstring CommonManaged::GetProductVersion()
+    {
+        return hstring{ get_product_version() };
+    }
+    winrt::Windows::Foundation::Collections::IVector<hstring> CommonManaged::GetAllActiveMicrophoneDeviceNames()
+    {
+        auto names = std::vector<winrt::hstring>();
+        for (const auto& device : MicrophoneDevice::getAllActive())
+        {
+            names.push_back(device->name().data());
+        }
+        return winrt::multi_threaded_vector(std::move(names));
```

Comment:
I kinda followed https://devblogs.microsoft.com/oldnewthing/20240522-00/?p=109795 .
Just thought it was safer too, since I'm not sure how WinUI3 in Settings will switch threads, so I thought I should try to keep it as broad as possible.